bundles/com.espressif.idf.launch.serial.ui/src/com/espressif/idf/launch/serial/ui/internal/NewSerialFlashTargetWizard.java

```diff
@@ -163,7 +163,7 @@ private boolean deleteDirectory(File directoryToBeDeleted) {
 				deleteDirectory(file);
 			}
 		}
-		return directoryToBeDeleted.delete();
+		return directoryToBeDeleted.getName().equals("build") ? true : directoryToBeDeleted.delete();
```

Comment:
Hi @sigmaaa 

I am not quite sure how this is helping here. Are we get into a case where we pass the non "build" directory to this method?

Could you replace this File delete recursive implementation with the Files.walkFileTree(..) and see if that helps here

```
Files.walkFileTree(directoryToBeDeleted, new SimpleFileVisitor<Path>() {
		   @Override
		   public FileVisitResult visitFile(Path file, BasicFileAttributes attributes) throws IOException {
		       Files.delete(file); 
		       return FileVisitResult.CONTINUE;
		   }

		   @Override
		   public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
		       Files.delete(dir); 
		       return FileVisitResult.CONTINUE;
		   }
		});
```

Reply:
Hi @kolipakakondal,

the problem in the delete function is that we are trying to delete the "build" directory and for some reason when we are trying to delete even an empty "build" directory after switching targets, there is a problem caused by locking this directory by Java. Then it's impossible to delete this directory or build the project once again (Only restart eclipse helps). Please check the following gif:

![IEP-317-demonstration](https://user-images.githubusercontent.com/24419842/105026591-fcca5700-5a4e-11eb-9f87-ae4a709c7502.gif)

To avoid this problem, I added a condition statement, that's checks if we are deleting the "build" directory and returns "true" just to skip deleting. Since we are deleting the "build" directory at the end of the delete process, it will leave an empty build directory at the end of the delete function. This is how it works with this statement: 

![IEP-317-demonstration2](https://user-images.githubusercontent.com/24419842/105027079-9db91200-5a4f-11eb-8afa-a6194961c04e.gif)

As you can see, there is an empty "build" directory after switching targets with no lock on it, so we can build the project with no error.

Once again, this issue happens only on Windows 10.